dask_cuda/device_host_file.py

```diff
@@ -76,18 +115,22 @@ def __init__(
 
     def __setitem__(self, key, value):
         if is_device_object(value):
-            self.device_buffer[key] = value
+            self.device_buffer[key] = _device_serialize(value)
```

Comment:
It looks like we're serializing objects as they arrive.  This seems odd to me.  I would only expect to serialize objects as we needed to move between one of the different memory spaces (host, device, disk).